apps/token-manager/app/src/screens/Holders.js

```diff
@@ -1,166 +1,171 @@
-                      items={TABS}
-                      selected={selectedTab}
-                      onSelect={this.handleSelectTab}
-                    />
-                  </TabBarWrapper>
-                )}
-                {(!tabbedNavigation || selectedTab === 0) && (
-                  <ResponsiveTable
-                    header={
-                      <TableRow>
-                        <TableHeader
-                          title={groupMode ? 'Owner' : 'Holder'}
-                          groupmode={groupMode}
-                          colSpan={groupMode ? '2' : '1'}
-                        />
-                        {!groupMode && (
-                          <TableHeader
-                            title="Balance"
-                            align={compactTable ? 'left' : 'right'}
-                            colSpan={compactTable ? '2' : '1'}
-                          />
-                        )}
-                        {!groupMode && !compactTable && <TableHeader />}
-                      </TableRow>
-                    }
-                    noSideBorders={compactTable}
-                  >
-                    {holders.map(({ address, balance }) => (
-                      <HolderRow
-                        key={address}
-                        address={address}
-                        balance={balance}
-                        groupMode={groupMode}
-                        isCurrentUser={Boolean(
-                          userAccount && userAccount === address
-                        )}
-                        maxAccountTokens={maxAccountTokens}
-                        tokenDecimalsBase={tokenDecimalsBase}
-                        onAssignTokens={onAssignTokens}
-                        onRemoveTokens={onRemoveTokens}
-                        compact={compactTable}
-                      />
-                    ))}
-                  </ResponsiveTable>
-                )}
-              </Main>
-              {(!tabbedNavigation || selectedTab === 1) && (
-                <SideBar
-                  holders={holders}
-                  tokenAddress={tokenAddress}
-                  tokenDecimalsBase={tokenDecimalsBase}
-                  tokenName={tokenName}
-                  tokenSupply={tokenSupply}
-                  tokenSymbol={tokenSymbol}
-                  tokenTransfersEnabled={tokenTransfersEnabled}
-                  userAccount={userAccount}
-                />
-              )}
-            </TwoPanels>
-          )
-        }}
-      </Viewport>
-    )
-  }
+            return values
+          }}
+          renderEntryActions={([address, balance]) => (
+            <EntryActions
+              address={address}
+              balance={balance}
+              onAssignTokens={onAssignTokens}
+              onRemoveTokens={onRemoveTokens}
+              singleToken={!balance || balance.eq(tokenDecimalsBase)}
+              canAssign={!!balance && balance.lt(maxAccountTokens)}
```

Comment:
I don’t remember discussing it, but I generally prefer to use `Boolean()`, `String()` and `Number()` for type conversions, rather than relying on the usual equivalents relying on type coercion rules (`!!`, `+""` and `-0`), for readability.

```suggestion
              canAssign={Boolean(balance) && balance.lt(maxAccountTokens)}
```

or

```suggestion
              canAssign={Boolean(balance && balance.lt(maxAccountTokens))}
```

Reply:
I've since simplified this logic and made it based on `groupMode` in 8ba0f83.